File: app/src/main/java/com/xpf/p2p/fragment/ProductHotFragment.java
package com.xpf.p2p.fragment;

import android.graphics.Color;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import com.loopj.android.http.RequestParams;
import com.xpf.p2p.R;
import com.xpf.p2p.base.BaseFragment;
import com.xpf.p2p.utils.DrawUtils;
import com.xpf.p2p.utils.UIUtils;
import com.xpf.p2p.widget.FlowLayout;

import java.util.Random;


/**
 * Created by xpf on 2016/11/15 :)
 * Function:热门理财页面
 */

public class ProductHotFragment extends BaseFragment {

    FlowLayout flowLayout;

    private String[] mData = new String[]{"新手计划", "乐享活系列90天计划", "钱包", "30天理财计划(加息2%)",
            "林业局投资商业经营与大捞一笔", "中学老师购买车辆", "屌丝下海经商计划", "新西游影视拍", "Java培训老师自己周转",
            "HelloWorld", "C++-C-ObjectC-java", "Android vs ios", "算法与数据结构", "JNI与NDK", "team working"};

    @Override
    public int getLayoutId() {
        return R.layout.fragment_product_hot;
    }

    @Override
    protected String getUrl() {
        return null;
    }

    @Override
    protected RequestParams getParams() {
        return null;
    }

    @Override
    protected void initData(String content) {
        flowLayout = (FlowLayout) mView.findViewById(R.id.flow_layout);
        Random random = new Random();

        for (String text : mData) {
            TextView tv = new TextView(getActivity());
            tv.setText(text);
            tv.setTextSize(UIUtils.dp2px(6));

            // 提供边距的对象,并设置到textView中
            ViewGroup.MarginLayoutParams mp = new ViewGroup.MarginLayoutParams(
                    ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
            mp.leftMargin = UIUtils.dp2px(8);
            mp.topMargin = UIUtils.dp2px(8);
            mp.rightMargin = UIUtils.dp2px(8);
            mp.bottomMargin = UIUtils.dp2px(8);

            tv.setLayoutParams(mp);

            // 设置背景
            int red = random.nextInt(210);
            int green = random.nextInt(210);
            int blue = random.nextInt(210);

            // 方式一：
            tv.setBackground(DrawUtils.getDrawable(Color.rgb(red, green, blue), UIUtils.dp2px(4)));

            // 方式二：
            tv.setBackground(DrawUtils.getSelector(DrawUtils.getDrawable(Color.rgb(red, green, blue),
                    UIUtils.dp2px(4)), DrawUtils.getDrawable(Color.WHITE, UIUtils.dp2px(4))));
            // 当设置了点击事件时,默认textView就是可点击的了
            tv.setOnClickListener(v -> Toast.makeText(
                    ProductHotFragment.this.getActivity(), tv.getText(), Toast.LENGTH_SHORT).show());

            // 设置内边距：
            int padding = UIUtils.dp2px(4);
            tv.setPadding(padding, padding, padding, padding);
            flowLayout.addView(tv);
        }
    }

}

File: app/src/main/java/com/xpf/p2p/fragment/ProductHotFragment.kt
package com.xpf.p2p.fragment

import android.graphics.Color
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast
import com.xpf.p2p.R
import com.xpf.p2p.base.BaseFragment
import com.xpf.p2p.databinding.FragmentProductHotBinding
import com.xpf.p2p.utils.DrawUtils
import com.xpf.p2p.utils.UIUtils
import java.util.Random

/**
 * Created by xpf on 2016/11/15 :)
 * Function:热门理财页面
 */
class ProductHotFragment : BaseFragment() {

    private var _binding: FragmentProductHotBinding? = null
    private val binding get() = _binding!!

    private val mData = arrayOf(
        "新手专享计划", "乐享活期90天", "零钱宝", "30天稳健计划(加息2%)",
        "小微企业经营周转", "个人消费信贷", "供应链金融计划", "影视文化产业基金", "教育分期项目",
        "智能定投组合", "季度增利计划", "年度稳盈计划", "灵活申赎产品", "新能源产业基金", "科技创新计划"
    )

    override fun getLayoutId(): Int = R.layout.fragment_product_hot

    override fun getUrl(): String = ""

    override fun getParams(): Map<String, String>? = null

    override fun initData(content: String?) {
        _binding = FragmentProductHotBinding.bind(mView!!)
        val random = Random()

        for (text in mData) {
            val tv = TextView(activity)
            tv.text = text
            tv.textSize = UIUtils.dp2px(6).toFloat()

            val mp = ViewGroup.MarginLayoutParams(
                ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT
            )
            mp.leftMargin = UIUtils.dp2px(8)
            mp.topMargin = UIUtils.dp2px(8)
            mp.rightMargin = UIUtils.dp2px(8)
            mp.bottomMargin = UIUtils.dp2px(8)

            tv.layoutParams = mp

            val red = random.nextInt(210)
            val green = random.nextInt(210)
            val blue = random.nextInt(210)

            tv.background = DrawUtils.getSelector(
                DrawUtils.getDrawable(Color.rgb(red, green, blue), UIUtils.dp2px(4).toFloat()),
                DrawUtils.getDrawable(Color.WHITE, UIUtils.dp2px(4).toFloat())
            )

            tv.setOnClickListener {
                Toast.makeText(activity, tv.text, Toast.LENGTH_SHORT).show()
            }

            val padding = UIUtils.dp2px(4)
            tv.setPadding(padding, padding, padding, padding)
            binding.flowLayout.addView(tv)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
